EgammaAnalysis/ElectronTools/plugins/CalibratedElectronProducersRun2.cc

```diff
@@ -49,12 +50,20 @@ CalibratedElectronProducerRun2T<T>::CalibratedElectronProducerRun2T( const edm::
   theEpCombinationTool(),
   theEnCorrectorRun2(theEpCombinationTool, conf.getParameter<bool>("isMC"), conf.getParameter<bool>("isSynchronization"), conf.getParameter<std::string>("correctionFile")),
   recHitCollectionEBToken_(consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("recHitCollectionEB"))),
-  recHitCollectionEEToken_(consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("recHitCollectionEE")))
+  recHitCollectionEEToken_(consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("recHitCollectionEE"))),
+  autoDataType(conf.getParameter<bool>("autoDataType"))
```

Comment:
can we make this the default configuration with something like : 

    autoDataType( (conf.existsAs<bool>("autoDataType")  && !conf.getParameter<bool>("autoDataType") ) ? 0 : 1 )